components/admin/shared/admin-access.provider.tsx

```
@@ -0,0 +1,59 @@
+'use client';
+
+import { createContext, useContext } from 'react';
+import type { AdminAccessContext } from '@/lib/types/admin/admin-access-context.type';
+
+/**
+ * React context for admin permission awareness.
+ * Provides role, permissions, and convenience flags to child components.
+ */
+const AdminAccessContext = createContext<AdminAccessContext | null>(null);
```

Biome (2.1.2)

[error] Shouldn't redeclare 'AdminAccessContext'. Consider to delete it or rename it.

'AdminAccessContext' is defined here:


(lint/suspicious/noRedeclare)